### PR TITLE
Validator

### PR DESCRIPTION
**Pull request by 0xnikter**
## Proper JSON 400 error on invalid account id

Related to the issue #16

### CODE REVIEW

**Comment by telezhnaya** [comment]
We still have 404 if we provide `.` as AccountId

**Comment by 0xnikter** [comment]
@telezhnaya just rebased on main

File: src/modules/coin/mod.rs [hunk 1]
```diff
@@ -1,32 +1,59 @@
...
 mod schemas;
 
+#[derive(serde::Serialize)]
+pub struct ValidationErrorJsonPayload {
+    pub message: String,
+    pub fields: Vec<String>,
+}
+
+/// Custom error handler
+impl From<&validator::ValidationErrors> for ValidationErrorJsonPayload {
```
Comment (telezhnaya) [inline src/modules/coin/mod.rs]: We already have error handler in main (line 100). I suggest not to touch anything here and add one more handler to main, near the previous one. And the implementation should be the same, it's also Validation error. This way, we can reuse it in all the modules

FYI
We do not use general approach here, we decided to go with rich errors as it's defined in Rosetta RPC
https://www.rosetta-api.org/docs/models/Error.html

File: src/modules/coin/resources.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use crate::{db_helpers, errors, modules, types};
 /// for the given timestamp/block_height.
 pub async fn get_near_balance(
     pool: web::Data<sqlx::Pool<sqlx::Postgres>>,
-    request: web::Path<schemas::BalanceRequest>,
+    request: actix_web_validator::Path<schemas::BalanceRequest>,
```
Comment (telezhnaya) [inline src/modules/coin/resources.rs:17]: Could you please make the same thing for all the endpoints? You need to change this file and `nft/resources.rs`

**Review by telezhnaya — CHANGES_REQUESTED** [review]
Great job!

I've added some comments, we need to cover all the endpoints with it.
Feel free to say "I'm tired of all this stuff", I'm ready to finish it for you

File: src/modules/coin/mod.rs (hunk 1 above)
Comment (0xnikter) [reply src/modules/coin/mod.rs]: Made some changes, could you recheck please?

File: src/modules/coin/resources.rs (hunk 2 above)
Comment (0xnikter) [reply src/modules/coin/resources.rs:17]: Done:)

File: src/modules/coin/mod.rs [hunk 3]
```diff
@@ -4,29 +4,36 @@ mod data_provider;
 mod resources;
 mod schemas;
 
+#[derive(serde::Serialize)]
+pub struct ValidationErrorJsonPayload {
+    pub message: String,
+    pub fields: Vec<String>,
+}
+
```
Comment (telezhnaya) [inline src/modules/coin/mod.rs]: Nit
We can drop this. Also, please run `cargo fmt`, there should be no changes in this file and in `nft/mod.rs`

**Review by telezhnaya — APPROVED** [review]
Awesome!

**Comment by telezhnaya** [comment]
Thank you @neeeekitos!

**Pull request merged**